ansible/roles/test/files/ptftests/sonic.py

```diff
@@ -0,0 +1,618 @@
+            if v6_routing != v6_routing_ok:
+                v6_routing_ok = v6_routing
+                self.log('BGP routing for ipv6 OK: %s' % (v6_routing_ok))
+            info["bgp_route_v6"] = v6_routing_ok
+
+            if not run_once:
+                self.ipv4_gr_enabled, self.ipv6_gr_enabled, self.gr_timeout = self.parse_bgp_neighbor_once(bgp_neig_output)
+                if self.gr_timeout is not None:
+                    log_first_line = "session_begins_%f" % cur_time
+                    self.do_cmd("send log message %s" % log_first_line)
+                    run_once = True
+
+            data[cur_time] = info
+            if self.DEBUG:
+                debug_data[cur_time] = {
+                    'show lacp neighbor' : lacp_output,
+                    'show ip bgp neighbors' : bgp_neig_output,
+                    'show ip route bgp' : bgp_route_v4_output,
+                    'show ipv6 route bgp' : bgp_route_v6_output,
+                }
+            time.sleep(1)
+
+        attempts = 20
+        end_time = time.time()
+        log_present = False
+        log_data = {}
+
+        self.log('Collecting logs')
+        log_lines = self.do_cmd("sudo cat /var/log/syslog /var/log/syslog.1 /var/log/teamd.log /var/log/teamd.log.1 /var/log/frr/bgpd.log /var/log/frr/zebra.log").split('\n')
+        syslog_regex_r = r'^(\S+\s+\d+\s+\d+:\d+:\d+)\.\d+ \S+ [A-Z]+ ([a-z\-]+#[/a-zA-Z0-9_]+)(?:\s+\d+-\d+-\d+\s+\d+:\d+:\d+,\d+\s+[A-Z]+\s+\w+)?(?:\[\d+\])?: (.+)$'
+        parsed_logs = self.extract_from_logs(syslog_regex_r, log_lines, min_timestamp=start_time)
+        self.log('Log output "{}"'.format('\n'.join(["{} {} {}".format(k[0], j, k[1]) for j in parsed_logs for k in parsed_logs[j]])))
+        log_data = self.parse_logs(parsed_logs)
+        if (self.reboot_type == 'fast-reboot' and \
+            'bgp#bgpd' in parsed_logs and 'PortChannel' in parsed_logs) \
+                or (self.reboot_type == 'warm-reboot' and 'bgp#bgpd' in parsed_logs) \
+                or (self.reboot_type == 'service-warm-restart' and 'bgp#bgpd' in parsed_logs):
+            log_present = True
+        else:
+            log_data['error'] = 'Incomplete logs'
+
+        self.log('Disconnecting from VM')
+        self.disconnect()
+
+        # save data for troubleshooting
+        with open("/tmp/%s.data.pickle" % self.ip, "w") as fp:
+            pickle.dump(data, fp)
+
+        # save debug data for troubleshooting
+        if self.DEBUG:
+            with open("/tmp/%s.raw.pickle" % self.ip, "w") as fp:
+                pickle.dump(debug_data, fp)
+            with open("/tmp/%s.logging" % self.ip, "w") as fp:
+                fp.write("\n".join(log_lines))
+
+        self.log('Checking BGP GR peer status on VM')
+        self.check_gr_peer_status(data)
+        cli_data = {}
+        #cli_data['lacp']   = self.check_series_status(data, "lacp",         "LACP session")
```

Comment:
Why not collect lacp data?

Reply:
I didn't realize that there was a command `show interface portchannel` that would fit well here. I'll add that in.